### DISCUSSION TITLE
How to freeze the scroll event on GLSP beyond specific zoom value

### DISCUSSION THREAD

**Discussion by praveen69sk** [eclipse-glsp/glsp#890] [Q&A]
Hi, 

I have the following the requirement:

1. When a user is scrolling the mouse wheel (Zoom In/Out) on the canvas, He should be able to zoom in/out.
2. Have set the thresholds betwee
![GLSP_Err](https://user-images.githubusercontent.com/103233516/215458115-f131b7a9-b761-4e80-884f-27e2e983a7a8.png)
n 2 to 200%, but when it reaches the Max threshold, scroll event is still happening and canvas view is going out of view. (Means it is scrolling to some other co-ordinates)

Requesting yours suggestions to limit the scroll event of the canvas, That should get freeze when the zoom values goes beyond the threshold values.

**Comment by planger** [comment]
Hi,

I think you would have to rebind a customized version of the [Sprotty's zoom listener](https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/features/viewport/zoom.ts) to do that and limit when and which `SetViewportAction` you are sending. Alternatively, you can also rebind and customize the handler of the `SetViewportAction`, which is the [`SetViewPortCommand`](https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/features/viewport/viewport.ts).

Both the zoom listener and the `SetViewportCommand` are bound here https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/features/viewport/di.config.ts.

**Reply by praveen69sk** [reply]
Thanks a lot! for the quick response. Will check this and let you know.
